ui/initial_state.py

```diff
@@ -19,20 +19,22 @@
     'replacing_library' : False,
     'backing_up_log'    : False,
     'shutting_down'     : False,
+    'halt_ui'           : False,
     'warming_up'        : False,
     'resetting_display' : False,
+    'update_ui'         : False,
     'display'           : frozendict({'width': 40, 'height': 9}),
 })
 
 state_file = 'state.pkl'
 
-def read():
-    log.debug('reading initial state')
+def read(state_file = state_file):
+    log.debug('reading initial state from %s' % state_file)
     try:
         with open(state_file) as fh:
             state = pickle.load(fh)
             return state
-    except:
+    except IOError:
```

Comment:
I really don't follow that logic. The higher up we go the more likely we are to swallow random exceptions which is the reason you should avoid blanket except clauses. 

But just looking at this case here, if any random error occurs when it is trying to acquire the state, why would it not be better to return the default state?